core_contracts/governance/governance.py

```diff
@@ -245,6 +335,7 @@ def executeVoteAction(self, vote_index: int) -> None:
                 except BaseException as e:
                     revert(f"Failed Execution of action. Reason: {e}")
```

Comment:
Coupled refunding to a successful execution of vote actions here. Not sure if you have something else in mind? Do we want a user to be refunded even if he makes a mistake when defining the actions?  One option could be to remove the revert statement here if we want the user to be refunded on a successful vote even if execution fails.

Reply:
One other thought about the process - We have this idea of activating a vote, or maybe we need to call it something else, and canceling a vote if there is something wrong with it. Do we want to allow that as part of the process? It should not include requiring the governance owner to approve. The activate or cancel would be called by the one who defined the vote. It would just give them the chance to back out if there was an error. If we aren't going to use those methods they could be removed. If they will be used we could make the payment happen in the activate, or approve, method. Would it even make sense to require an approver who is also qualified to submit a vote? Like a code review?

Reply:
I see Scott just made a comment to remove the activateVote methods; so I guess I'll do that. I'll also add refunding to the cancelVote function along with access to the proposer.

Actually I see some problems with the executeVoteAction method. A proposals vote actions can be called repeatedly to execute already executed vote actions (as far as I can tell). Someone might call it on an old proposal that has already been executed and change some really vital stuff. I'll address this.

Another thing. You'll have two different kind of votes - votes that require execution and votes that are only "polls", like the one currently ongoing. But the proposal status "successful" is never asigned in any senario. I'll fix that too.

Please let me know if I misunderstood something.

Reply:
Had to make changed all over to make it all fit together .. sorry about that.

Reply:
Optionally we could remove activateVote and activate the vote on creation, but allow it to be cancelled as long as it has not started yet. I was also thinking about "malicious" proposals. Like if you submit a proposal with clearly malicious parameter changes, payout of funds from treasury and such, but with a completely different description to trick users into voting yes. Very hard to spot for the average user. Would it be good to perhaps limit vote_start to >= current_day + 3 (or something like that). So there is ample time for someone to catch those and inform?